specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,133 @@
+# Table Column [Template]
+
+*Refer to the [table column type philosophy docs](/packages/nimble-components/src/table/specs/table-columns-hld.md#column-type-philosophy) for column creation policies and naming conventions.*
+
+## Overview
+
+*A high-level description of the column.*
+
+### Background
+
+- *Relevant historical or background information*
+- *Link to relevant work items, related existing issues, etc.*
+
+### Features
+
+*A list of the key features unique to this component. If adding support for an existing Nimble component's capabilities (e.g. as `nimble-table-column-mapping` does for Nimble icons and `nimble-spinner`), consider which parts of that component's API should be exposed, and whether on a per-column or per-cell basis. Any parts of the API explicitly not supported should be called out below as non-goals.*
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+
+## Implementation / Design
+
+*Describe interesting implementation/design decisions, with justifications and alternates considered.*
+
+*Include code snippets showing basic component use and any interesting configurations.*
+
+Typical usage:
+
+```html
+<nimble-table>
+    <nimble-table-column-*>
+        Column Label
+    </nimble-table-column-*>
+</nimble-table>
+```
+
+### API
+
+#### Column Element
+
+_Element Name_
+
+-   `nimble-table-column-*`
```

Comment:
We should either link to or move the column type philosophy stuff here: https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/specs/table-columns-hld.md#column-type-philosophy

Reply:
There's a link to it at the top of the file. It will be essentially the first thing someone filling out the template will see (and then delete). There's no need to also link it from here as well, right?

Reply:
Moved the existing link to right before the element name section.